### DISCUSSION TITLE
Advanced Config | Mapping Config | Backup & Restore

### DISCUSSION THREAD

**Discussion by Johndev5** [darthcloud/BlueRetro#1109] [Ideas]
Hi @darthcloud,

I absolutely love what you've done with BlueRetro, I've not long bought a device for the GameCube/Wii and started looking at mapping for a bunch of games so we can play party games with little to no messing around for them.

I think I've seen on another platform you mentioning that there are a limited amount of mapping presets that can be stored on the devices themselves however would you consider allowing the ability for single device presets on the Advanced Config > Mapping Config page (i.e. all of the manual mapping that was on that page) to be saved to a text editable file (presumably XML or a generic text based config file) then a function for any device number selected in the option list on that page to then import that preset to that device?

I've already solved this problem for myself using Power Automate Desktop via quite a simple means (Currently just to switch between the default config and one specifically for CTGPR). I just thought it would be a useful feature to allow for off device profile backup & restore.

**Comment by darthcloud** [comment]
I guess the main issue is that I despise working on the web interface. I agree it would be usefull.

**Reply by Johndev5** [reply]
Haha fair enough, I've tinkered with Javascript so I could probable create a Chrome extension to add a button and maybe another option list to the bottom of the page to switch between the CTGPR config and the default config.

For me specifically it's the (1) dualsense/dualshock 4 (3) controllers. (I've tried a few times with Stadia but for whatever reason there's always an issue that I can't pin down.

If I do get that extension I'll post here if I get it done. (Assuming this is an appropriate place for it.)

**Reply by darthcloud** [reply]
why an extension, the site is open source:
https://github.com/darthcloud/BlueRetrowebcfg

Fell free to send a PR

**Reply by Johndev5** [reply]
Mainly becase what I'm trying to do seems quite niche compared to the breadth of devices and games this awesome piece of software works with, that and I'm quite rusty on the basic web dev skills I learnt.

PR == Push request? (Rusty on GIT too)

I haven't even got around to finishing the extension at this point.
